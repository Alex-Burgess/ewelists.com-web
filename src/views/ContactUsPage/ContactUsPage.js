import React, { useState } from 'react';
// libs
import { onError, debugError } from "libs/errorLib";
import { contactRequest } from "libs/apiLib";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
// core components
import GridContainer from "components/Grid/GridContainer.js";
import GridItem from "components/Grid/GridItem.js";
import Input from "components/Input/CustomInput.js";
import Button from "components/Buttons/Button.js";
import Footer from "components/Footer/FooterGrey.js";
import HeaderWhite from "components/Header/HeaderWhite.js";
import Card from "components/Card/Card.js";
import CardBody from "components/Card/CardBody.js";
import ErrorText from "components/Typography/Error.js";

import styles from "assets/jss/material-kit-pro-react/views/contactPageStyle.js";
const useStyles = makeStyles(styles);

export default function ContactUsPage(props) {
  const classes = useStyles();
  const [submit, setSubmit] = useState(false);
  const [name, setName] = useState('');
  const [email, setEmail] = useState('');
  const [message, setMessage] = useState('');
  const [error, setError] = useState('');

  const sendMail = async event => {
    let details = {
      name: name,
      email: email,
      message: message
    }

    debugError("API request body: " + JSON.stringify(details));

    try {
      await contactRequest(details);
      setSubmit(true);
    } catch (e) {
      onError(e);
      setError("There was an unexpected error, we're working on it!");
    }
  }

  const validateForm = () => {
    return (
      name.length > 0 &&
      email.length > 0 &&
      message.length > 0
    );
  }


  const renderForm = () => {
    return (
      <form>
        <Input
          labelText="Your Name"
          id="name"
          formControlProps={{
            fullWidth: true
          }}
          inputProps={{
            onChange: event => setName(event.target.value)
          }}
        />
        <Input
          labelText="Email address"
          id="email"
          formControlProps={{
            fullWidth: true
          }}
          inputProps={{
            onChange: event => setEmail(event.target.value)
          }}
        />
        <Input
          labelText="Your message"
          id="message"
          formControlProps={{
            fullWidth: true
          }}
          inputProps={{
            multiline: true,
            rows: 6,
            onChange: event => setMessage(event.target.value)
          }}
        />
        <div className={classes.textCenter}>
          <Button color="primary" round onClick={() => sendMail()} disabled={!validateForm()}>
            Contact us
          </Button>
          {error
            ? <div className={classes.textCenter}>
                <ErrorText>
                  <p>{error}</p>
                </ErrorText>
              </div>
            : null
          }
        </div>
      </form>
    )
  }


  return (
    <div className={classes.page}>
      <HeaderWhite isAuthenticated={props.isAuthenticated} user={props.user} mobile={props.mobile} />
      <div className={classes.container}>
        <GridContainer justify="center">
          <GridItem xs={12} sm={10} md={8} className={classes.gridLogin}>
            <Card className={classes.customCard}>
              <CardBody signup>
                <h3 className={classes.title + " " + classes.textCenter}>Send us a message</h3>
                <h4>
                  You can contact us with anything related to Ewelists. We {"'"}ll get back to you ASAP.
                </h4>
                <div id="formWrapper">
                  {submit
                    ? <div className={classes.sent}>
                        Thank you for your message {name}.  We aim to respond to emails the same day.
                      </div>
                    : renderForm()
                  }
                </div>
              </CardBody>
            </Card>
          </GridItem>
        </GridContainer>
      </div>
      <div className={classes.flexer} />
      <Footer />
    </div>
  );
}
