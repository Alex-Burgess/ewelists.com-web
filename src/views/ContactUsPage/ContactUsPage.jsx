/*!

=========================================================
* Material Kit PRO React - v1.7.0
=========================================================

* Product Page: https://www.creative-tim.com/product/material-kit-pro-react
* Copyright 2019 Creative Tim (https://www.creative-tim.com)

* Coded by Creative Tim

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

*/
/*eslint-disable*/
import React from "react";
import { API } from "aws-amplify";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// nodejs library that concatenates classes
import classNames from "classnames";
// @material-ui/core components
import withStyles from "@material-ui/core/styles/withStyles";
// @material-ui/icons
import Favorite from "@material-ui/icons/Favorite";
import PinDrop from "@material-ui/icons/PinDrop";
import Phone from "@material-ui/icons/Phone";
import BusinessCenter from "@material-ui/icons/BusinessCenter";
// core components
import Header from "components/Header/Header.jsx";
import HeaderLinks from "components/Header/HeaderLinks.jsx";
import HeaderLinksAuth from "components/Header/HeaderLinksAuth.jsx";
import GridContainer from "components/Grid/GridContainer.jsx";
import GridItem from "components/Grid/GridItem.jsx";
import InfoArea from "components/InfoArea/InfoArea.jsx";
import CustomInput from "components/CustomInput/CustomInput.jsx";
import Button from "components/CustomButtons/Button.jsx";
import FooterDark from "components/Footer/FooterDark.jsx";
import Parallax from "components/Parallax/Parallax.jsx";

import contactUsStyle from "assets/jss/material-kit-pro-react/views/contactUsStyle.jsx";

class ContactUsPage extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      name: '',
      email: '',
      message: '',
      submit: false
    };
  }

  sendMail = async event => {
    let details = {
      name: this.state.name,
      email: this.state.email,
      message: this.state.message
    }
    const response = await API.post("contact", "/", {
      body: details
    });

    console.log("response: " + JSON.stringify(response))
    this.setState({
      submit: true
    });
  }

  changeHandler = event => {
    this.setState({
      [event.target.id]: event.target.value
    })
  }

  validateForm() {
    return (
      this.state.name.length > 0 &&
      this.state.email.length > 0 &&
      this.state.message.length > 0
    );
  }


  renderForm(classes){
    return (
      <form>
        <CustomInput
          labelText="Your Name"
          id="name"
          formControlProps={{
            fullWidth: true
          }}
          inputProps={{
            onChange: this.changeHandler
          }}
        />
        <CustomInput
          labelText="Email address"
          id="email"
          formControlProps={{
            fullWidth: true
          }}
          inputProps={{
            onChange: this.changeHandler
          }}
        />
        <CustomInput
          labelText="Your message"
          id="message"
          formControlProps={{
            fullWidth: true
          }}
          inputProps={{
            multiline: true,
            rows: 6,
            onChange: this.changeHandler
          }}
        />
        <div className={classes.textCenter}>
          <Button color="primary" round onClick={() => this.sendMail()} disabled={!this.validateForm()}>
            Contact us
          </Button>
        </div>
      </form>
    )
  }

  renderSent(classes){
    return (
      <div className={classes.sent}>
        <p>
          Thank you for your message {this.state.name}.  We will get back to you by email.
        </p>
      </div>
    )
  }

  render() {
    const { classes } = this.props;
    return (
      <div>
        {this.props.isAuthenticated
          ? <Header
              brand="ewelists"
              links={<HeaderLinksAuth dropdownHoverColor="info" />}
              fixed
              color="info"
            />
          : <Header
              brand="ewelists"
              links={<HeaderLinks dropdownHoverColor="info" />}
              fixed
              color="info"
            />
        }
        <Parallax image={require("assets/img/sheep-with-shoes.jpg")} className={classes.parallax}>
          <div className={classes.container}>
            <GridContainer justify="center">
              <GridItem
                md={8}
                sm={8}
                className={classNames(
                  classes.mlAuto,
                  classes.mrAuto,
                  classes.textCenter
                )}
              >
              </GridItem>
            </GridContainer>
          </div>
        </Parallax>
        <div className={classNames(classes.main, classes.mainRaised)}>
          <div className={classes.contactContent}>
            <div className={classes.container}>
              <h2 className={classes.title}>Send us a message</h2>
              <GridContainer>
                <GridItem md={12} sm={12}>
                  <p className={classes.subHeading}>
                    You can contact us with anything related to Ewelists. We
                    {"'"}ll get in touch with you as soon as possible.
                    <br />
                    <br />
                  </p>
                  {this.state.submit
                    ? this.renderSent(classes)
                    : this.renderForm(classes)
                  }
                </GridItem>
              </GridContainer>
            </div>
          </div>
        </div>
        <FooterDark />
      </div>
    );
  }
}

ContactUsPage.propTypes = {
  classes: PropTypes.object
};

export default withStyles(contactUsStyle)(ContactUsPage);
