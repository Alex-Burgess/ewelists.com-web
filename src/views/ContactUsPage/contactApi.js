import { API } from "aws-amplify";

export function contactApiPost(details) {
  return API.post("contact", "/", {
    body: details
  });
}
