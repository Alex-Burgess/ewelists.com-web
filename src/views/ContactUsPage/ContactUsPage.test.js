import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import {mount} from 'enzyme';
import { act } from 'react-dom/test-utils';
import ContactUsPage from './ContactUsPage';

describe('ContactUs Page Snapshots', () => {
  test('Renders empty form of contact us page', () => {
    const tree = TestRenderer
      .create(<Router><ContactUsPage /></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Test Input Form', () => {
  const wrapper = mount(
    <Router>
      <ContactUsPage />
    </Router>
  );

  it('Should have 1 form', () => {
    expect(wrapper.find('form').length).toEqual(1);
  });

  it('Should have 3 inputs, name, email and message', () => {
    expect(wrapper.find('CustomInput').length).toEqual(3);
    expect(wrapper.find('CustomInput[id="name"]').length).toEqual(1);
    expect(wrapper.find('CustomInput[id="email"]').length).toEqual(1);
    expect(wrapper.find('CustomInput[id="message"]').length).toEqual(1);
  });

  it('Should have 1 form submit button, which is disabled', () => {
    const submitButton = wrapper.find('form').find('button');
    expect(submitButton.length).toEqual(1);
    expect(submitButton.hasClass(/(makeStyles-disabled)-(\d+)/)).toEqual(true);
  });

  it('Submit button should be active when all fields populated', () => {
    wrapper.find('input[id="name"]').simulate('change', {
      target: {
        value: 'Test User',
      },
    });

    wrapper.update();
    var submitButton = wrapper.find('form').find('button');
    expect(submitButton.hasClass(/(makeStyles-disabled)-(\d+)/)).toEqual(true);

    wrapper.find('input[id="email"]').simulate('change', {
      target: {
        value: 'test.user@gmail.com',
      },
    });

    wrapper.update();
    submitButton = wrapper.find('form').find('button');
    expect(submitButton.hasClass(/(makeStyles-disabled)-(\d+)/)).toEqual(true);

    wrapper.find('textarea[id="message"]').simulate('change', {
      target: {
        value: 'A test message.',
      },
    });

    wrapper.update();
    submitButton = wrapper.find('form').find('button');
    // console.log("Button: " + submitButton.debug());
    expect(submitButton.hasClass(/(makeStyles-disabled)-(\d+)/)).toEqual(false);
  });
});


describe('Test Email Sent Result Page', () => {
  const api = require('./contactApi.js');
  api.contactApiPost = jest.fn(() => {
    return {
      "name": 'Mock User',
      "email": 'mock.user@gmail.com',
      "message": 'A test message.',
      "id": 258166
    }
  });

  const wrapper = mount(
    <Router>
      <ContactUsPage />
    </Router>
  );

  it('Should send email and display confirmation message', async () => {
    wrapper.find('input[id="name"]').simulate('change', {
      target: {
        value: 'Test User',
      },
    });

    wrapper.find('input[id="email"]').simulate('change', {
      target: {
        value: 'test.user@gmail.com',
      },
    });

    wrapper.find('textarea[id="message"]').simulate('change', {
      target: {
        value: 'Test Message',
      },
    });

    wrapper.update();
    expect(wrapper.find('#formWrapper').length).toEqual(1);

    expect(wrapper.find('#formWrapper').text()).toEqual('Your NameEmail addressYour messageContact us');
    // expect(wrapper.find('#formWrapper').html()).toEqual('Your NameEmail addressYour messageContact us');

    const submitButton = wrapper.find('form').find('button');
    // submitButton.simulate('click');

    await act(async () => {
      submitButton.simulate('click');
    });

    expect(wrapper.find('#formWrapper').text()).toEqual('Thank you for your message Test User.  We will get back to you by email.');
  });
});
