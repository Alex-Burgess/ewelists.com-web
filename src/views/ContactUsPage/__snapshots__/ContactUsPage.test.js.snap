// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUs Page Snapshots Renders empty form of contact us page 1`] = `
<div>
  <div>
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-41 makeStyles-info-48 makeStyles-fixed-43 MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular makeStyles-container-44 MuiToolbar-gutters"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-title-45"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <img
                alt="logo"
                className="makeStyles-logo-40"
                src="logo-white.png"
              />
            </a>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <div
          className="makeStyles-hidden-57 PrivateHiddenCss-smDown-139"
        >
          <div
            className="makeStyles-collapse-58"
          >
            <ul
              className="MuiList-root makeStyles-list-149 makeStyles-mlAuto-165 MuiList-padding"
            >
              <li
                className="MuiListItem-root makeStyles-listItem-150 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-navLink-152"
                  href="/listideas"
                  onClick={[Function]}
                >
                  Gift List Ideas
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-listItem-150 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-navLink-152"
                  href="/login"
                  onClick={[Function]}
                >
                  Log In
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-listItem-150 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-navLink-152"
                  href="/signup"
                  onClick={[Function]}
                >
                  Sign Up
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </header>
  </div>
  <div
    className="makeStyles-parallax-181 makeStyles-parallax-18"
    style={
      Object {
        "backgroundImage": "url(sheep-with-shoes.jpg)",
        "transform": "translate3d(0,0px,0)",
      }
    }
  >
    <div
      className="makeStyles-container-6"
    >
      <div
        className="MuiGrid-root makeStyles-grid-191  MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          className="MuiGrid-root makeStyles-grid-295 makeStyles-mlAuto-4 makeStyles-textCenter-10 MuiGrid-item MuiGrid-grid-sm-8 MuiGrid-grid-md-8"
        />
      </div>
    </div>
  </div>
  <div
    className="makeStyles-main-1 makeStyles-mainRaised-2"
  >
    <div
      className="makeStyles-contactContent-7"
    >
      <div
        className="makeStyles-container-6"
      >
        <h2
          className="makeStyles-title-3"
        >
          Send us a message
        </h2>
        <div
          className="MuiGrid-root makeStyles-grid-191  MuiGrid-container"
        >
          <div
            className="MuiGrid-root makeStyles-grid-295  MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
          >
            <p
              className="makeStyles-subHeading-19"
            >
              You can contact us with anything related to Ewelists. We
              '
              ll get in touch with you as soon as possible.
              <br />
              <br />
            </p>
            <div
              id="formWrapper"
            >
              <form>
                <div
                  className="MuiFormControl-root makeStyles-formControl-304 MuiFormControl-fullWidth"
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root makeStyles-labelRoot-300  MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink={false}
                    htmlFor="name"
                  >
                    Your Name
                  </label>
                  <div
                    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-underline-297 MuiInputBase-formControl MuiInput-formControl"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      className="MuiInputBase-input MuiInput-input makeStyles-input-306"
                      disabled={false}
                      id="name"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="text"
                    />
                  </div>
                </div>
                <div
                  className="MuiFormControl-root makeStyles-formControl-304 MuiFormControl-fullWidth"
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root makeStyles-labelRoot-300  MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink={false}
                    htmlFor="email"
                  >
                    Email address
                  </label>
                  <div
                    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-underline-297 MuiInputBase-formControl MuiInput-formControl"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      className="MuiInputBase-input MuiInput-input makeStyles-input-306"
                      disabled={false}
                      id="email"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="text"
                    />
                  </div>
                </div>
                <div
                  className="MuiFormControl-root makeStyles-formControl-304 MuiFormControl-fullWidth"
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root makeStyles-labelRoot-300  MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink={false}
                    htmlFor="message"
                  >
                    Your message
                  </label>
                  <div
                    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-underline-297 MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
                    onClick={[Function]}
                  >
                    <textarea
                      aria-invalid={false}
                      className="MuiInputBase-input MuiInput-input makeStyles-input-306 MuiInputBase-inputMultiline MuiInput-inputMultiline"
                      disabled={false}
                      id="message"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      rows={6}
                    />
                  </div>
                </div>
                <div
                  className="makeStyles-textCenter-10"
                >
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-366 makeStyles-primary-368 makeStyles-round-393 makeStyles-disabled-390"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      Contact us
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="makeStyles-spacer-17"
  />
  <div>
    <div />
    <footer
      className="makeStyles-footer-409 makeStyles-footer-404"
    >
      <div
        className="makeStyles-container-415"
      >
         
        <div>
          <div
            className="makeStyles-left-401"
          >
            <ul
              className="MuiList-root makeStyles-list-400 MuiList-padding"
            >
              <li
                className="MuiListItem-root makeStyles-inlineBlock-399 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-398"
                  href="/"
                >
                  © 
                  2020
                   ewelists
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-inlineBlock-399 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-398"
                  href="/contact"
                >
                  Contact Us
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-inlineBlock-399 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-398"
                  href="/#about"
                >
                  About us
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-inlineBlock-399 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-398"
                  href="/privacy"
                >
                  Privacy
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-inlineBlock-399 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-398"
                  href="/terms"
                >
                  Terms
                </a>
              </li>
            </ul>
          </div>
          <div
            className="makeStyles-rightLinks-405"
          >
            <ul>
              <li>
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-366 makeStyles-facebook-377 makeStyles-simple-388 makeStyles-justIcon-396"
                  href="https://www.facebook.com/ewelists"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  target="_blank"
                >
                  <span
                    className="MuiButton-label"
                  >
                    <i
                      className="fab fa-facebook"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </a>
              </li>
              <li>
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-366 makeStyles-twitter-376 makeStyles-simple-388 makeStyles-justIcon-396"
                  href="https://twitter.com/ewelists"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  target="_blank"
                >
                  <span
                    className="MuiButton-label"
                  >
                    <i
                      className="fab fa-twitter"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </a>
              </li>
              <li>
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-366 makeStyles-instagram-387 makeStyles-simple-388 makeStyles-justIcon-396"
                  href="https://www.instagram.com/ewelists/"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  target="_blank"
                >
                  <span
                    className="MuiButton-label"
                  >
                    <i
                      className="fab fa-instagram"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          className="makeStyles-clearFix-425"
        />
      </div>
    </footer>
  </div>
</div>
`;
